File: 202308/2483.go
func bestClosingTime(customers string) int {
	n := len(customers)
	ans := n
	cnt := 0

	for i := n - 1; i >= 0; i-- {
		if customers[i] == 'Y' {
			cnt += 1
		} else {
			cnt -= 1
		}
		if cnt <= 0 {
			ans = i
			cnt = 0
		}
	}
	return ans
}

File: 202512/2483.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        curr = customers.count("Y")
        n = len(customers)
        ans = 0
        ansv = curr
        for i in range(1, n + 1):
            if customers[i - 1] == "Y":
                curr -= 1
            else:
                curr += 1
            if ansv > curr:
                curr = ansv
                ans = i
        return ans
